packages/common-scripts/src/anyone_can_pay.ts

```diff
@@ -0,0 +1,549 @@
+      template.HASH_TYPE === input.cell_output.lock.hash_type &&
+      !processedArgs.has(input.cell_output.lock.args)
+    ) {
+      processedArgs = processedArgs.add(input.cell_output.lock.args);
+
+      // skip if input sum capcity <= output sum capacity
+      // and input sum amount <= output sum amount
+      const outputs: List<Cell> = txSkeleton.get("outputs").filter((output) => {
+        return new ScriptValue(output.cell_output.lock, {
+          validate: false,
+        }).equals(new ScriptValue(input.cell_output.lock, { validate: false }));
+      });
+      const sumOfOutputCapacity: bigint = outputs
+        .map((output) => BigInt(output.cell_output.capacity))
+        .reduce((result, c) => result + c, 0n);
+
+      const sumOfOutputAmount: bigint = outputs
+        .filter((output) => output.data !== "0x")
+        .map((output) => readBigUInt128LE(output.data))
+        .reduce((result, c) => result + c, 0n);
+
+      const fInputs: List<Cell> = inputs.filter((i) => {
+        return new ScriptValue(i.cell_output.lock, { validate: false }).equals(
+          new ScriptValue(input.cell_output.lock, { validate: false })
+        );
+      });
+
+      const sumOfInputCapacity: bigint = fInputs
+        .map((i) => BigInt(i.cell_output.capacity))
+        .reduce((result, c) => result + c, 0n);
+
+      const sumOfInputAmount: bigint = fInputs
+        .filter((i) => i.data !== "0x")
+        .map((i) => readBigUInt128LE(i.data))
+        .reduce((result, c) => result + c, 0n);
+
+      if (
+        sumOfInputCapacity <= sumOfOutputCapacity &&
+        sumOfInputAmount <= sumOfOutputAmount
+      ) {
+        continue;
+      }
+
+      const lockValue = new values.ScriptValue(input.cell_output.lock, {
+        validate: false,
+      });
+      const hasher = new CKBHasher();
+      hasher.update(txHash);
+      if (i >= witnesses.size) {
+        throw new Error(
+          `The first witness in the script group starting at input index ${i} does not exist, maybe some other part has invalidly tampered the transaction?`
+        );
+      }
+      hashWitness(hasher, witnesses.get(i)!);
+      for (let j = i + 1; j < inputs.size && j < witnesses.size; j++) {
+        const otherInput = inputs.get(j)!;
+        if (
+          lockValue.equals(
+            new values.ScriptValue(otherInput.cell_output.lock, {
+              validate: false,
+            })
+          )
+        ) {
+          hashWitness(hasher, witnesses.get(j)!);
+        }
+      }
+      for (let j = inputs.size; j < witnesses.size; j++) {
+        hashWitness(hasher, witnesses.get(j)!);
+      }
+      const signingEntry = {
+        type: "witness_args_lock",
+        index: i,
+        message: hasher.digestHex(),
+      };
+      signingEntries = signingEntries.push(signingEntry);
+    }
+  }
+  txSkeleton = txSkeleton.set("signingEntries", signingEntries);
+  return txSkeleton;
+}
+
+export async function destroyAnyoneCanPayInput(
```

Comment:
One we change transfer to withdraw, I feel like we can merge this function with withdraw: the cell will be destroyed once one withdraw all the amount in the cell.

Reply:
Updated.